### PR TITLE
adds citation file

### LINKED ISSUE #1004: Add citation file
## 🚀 Feature
We should add a citation file using the [citation file format](https://citation-file-format.github.io/). We could potentially use a citation format that is provided by CiteAs - see here http://citeas.org/cite/https://github.com/napari/napari for example (i'm not saying we want to use that exact one, but linking to a place which autogenerates them). I do like having a dedicated file though.

### PR DESCRIPTION
**Pull request by neuromusic**
# Description

adds a `CITATION.cff`-formatted citation file to the napari repo. 

this standard format will enable

- Github to render "How to cite" on the sidebar
- Zenodo will read this file to populate authors, etc in the "official" archived version of each release

citation generated from current zenodo metadata, using the [doi2cff](https://doi2cff.readthedocs.io/en/latest/) command line utility

## Type of change

housekeeping/documentation

# References

closes #1004

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:8]: @sofroniewn can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:12]: @tlambert03 can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:16]: @kne42 can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:19]: @jni can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:22]: @Czaki can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:26]: @pwinston can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:29]: @goanpeca can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:32]: @kevinyamauchi can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:35]: @Carreau can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:39]: @DragaDoncila can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff]: @AhmetCanSolak can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff]: @potating-potato can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:47]: @ppwadhwa can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff]: @alisterburt can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:53]: @GenevieveBuckley can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff]: @andy-sweet can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:60]: @lukasz-migas can you review your name & affiliation?

File: CITATION.cff
Comment (sofroniewn) [inline CITATION.cff]:
```suggestion
  affiliation: Chan Zuckerberg Initiative
```

File: CITATION.cff
Comment (sofroniewn) [reply CITATION.cff:8]: Name looks good, I suggested expanding CZI to Chan Zuckerberg Initiative

File: CITATION.cff
Comment (DragaDoncila) [inline CITATION.cff]:
```suggestion
- given-names: Draga
  family-names: Doncila Pop
```

File: CITATION.cff
Comment (DragaDoncila) [reply CITATION.cff:39]: @neuromusic looks fine, just tweaked the family names - had to look up what a `name-particle` was so TIL!

File: CITATION.cff
Comment (lukasz-migas) [reply CITATION.cff:60]: That looks good to me! 👍

File: CITATION.cff
Comment (Czaki) [inline CITATION.cff]:
```suggestion
  affiliation: University of Warsaw, Faculty of Mathematics, Informatics, and Mechanics
```

**Review by Czaki — CHANGES_REQUESTED** [review]

File: CITATION.cff
Comment (jni) [reply CITATION.cff:19]: It's good!

File: CITATION.cff
Comment (alisterburt) [inline CITATION.cff]:
```suggestion
- given-names: Alister
  family-names: Burt
  affiliation: MRC-LMB
```

Comment (alisterburt) [reply CITATION.cff]: thanks @neuromusic - hope this looks right!

File: CITATION.cff
Comment (kevinyamauchi) [reply CITATION.cff:32]: Looks good to me! Thanks, @neuromusic !

Comment (neuromusic) [reply CITATION.cff:32]: thank you!

File: CITATION.cff
Comment (tlambert03) [inline CITATION.cff]:
```suggestion
  affiliation: Harvard Medical School
```

File: CITATION.cff
Comment (neuromusic) [reply CITATION.cff:39]: oh wow, fascinating! yeah, this was what the `doi2cff` tool generated from the zenodo entry

File: CITATION.cff
Comment (neuromusic) [reply CITATION.cff:60]: awesome! thanks!

File: CITATION.cff
Comment (neuromusic) [reply CITATION.cff]: thanks!

File: CITATION.cff
Comment (neuromusic) [reply CITATION.cff:22]: thanks!

File: CITATION.cff
Comment (neuromusic) [reply CITATION.cff:12]: thanks!

File: CITATION.cff
Comment (andy-sweet) [inline CITATION.cff]:
```suggestion
- given-names: Andrew
  family-names: Sweet
  affiliation: Chan Zuckerberg Initiative
```

Probably makes sense to be consistent about the CZI affiliation across all contributors. Probably Chan Zuckerberg Initiative like @sofroniewn suggested, or maybe CZI or @chanzuckerberg.

File: CITATION.cff
Comment (goanpeca) [reply CITATION.cff:29]: Yes sir!

**Review by goanpeca — APPROVED** [review]

**Review by sofroniewn — APPROVED** [review]

File: CITATION.cff
Comment (pwinston) [reply CITATION.cff:26]: - given-names: Philip
  family-names: Winston
  affiliation: Tobeva Software

**Comment by sofroniewn** [comment]
> adds a CITATION.cff-formatted citation file to the napari repo.

@neuromusic are there recommendations on best practices to keep this file up to date over time? Also did you use a tool to autogenerate this initial one. Apologies if I missed link, but would be great to have, at least for historical record.

Awesome to get this going though. There's a lot of momentum in the community to make software citation easier and it's great to see napari joining in too!!

File: CITATION.cff
Comment (liu-ziyang) [reply CITATION.cff]: Given name: Ziyang
Family name: Liu
Affiliation: Chan Zuckerberg Initiative Foundation

File: CITATION.cff
Comment (liu-ziyang) [inline CITATION.cff]:
```suggestion
- given-names: Ziyang
  family-names: Liu
  affiliation: Chan Zuckerberg Initiative Foundation
```

File: CITATION.cff
Comment (Czaki) [inline CITATION.cff:24]:
```suggestion
- given-names: Grzegorz
  family-names: Bokota
  affiliation: University of Warsaw, Faculty of Mathematics, Informatics, and Mechanics
  orcid: https://orcid.org/0000-0002-5470-1676
```

**Review by Czaki — CHANGES_REQUESTED** [review]

**Comment by Czaki** [comment]
I think that it may be worth adding also orcid https://citation-file-format.github.io/

**Comment by neuromusic** [comment]
> @sofroniewn: did you use a tool to autogenerate this initial one.

yes: https://doi2cff.readthedocs.io/en/latest/ to generate the CFF file from the Zenodo DOI

> are there recommendations on best practices to keep this file up to date over time? 

I'm not sure... this author list was automatically generated by Zenodo... apparently with metadata from Github re: contributors to the git history (which does not include contributors to documentation or other repos like the plugin engine).

Once this PR lands, zenodo will begin to rely on this file instead of generating the citation (and author list) automagically... so you are exactly right that there will need to be some additional work to ensure that new contributors are included in the author list. This will probably mean more process and possibly a mechanism to include contributors who aren't represented in the author list b/c their contributions have been non-code, like @liaprins-czi  #3420 might be relevant here.

**Comment by sofroniewn** [comment]
> yes: https://doi2cff.readthedocs.io/en/latest/ to generate the CFF file from the Zenodo DOI

thanks!!

> there will need to be some additional work to ensure that new contributors are included in the author list. 

we could try and update/ regenerate before every release if easy/ automated

> This will probably mean more process and possibly a mechanism to include contributors who aren't represented in the author list b/c their contributions have been non-code, like @liaprins-czi #3420 might be relevant here.

agreed, thanks for including link here too

**Comment by sofroniewn** [comment]
Looks like this is coming together nicely. Maybe @neuromusic you and I can do a final clean up pass through this and get this in. I don't think it's urgent, but might as well keep it moving along. Would be nice to get a few more approvals from @napari/core-devs just to make sure everyone is on board with this approach before merge too!! Certainly i'd like an approval from @jni

File: CITATION.cff
Comment (tlambert03) [inline CITATION.cff]: i assume this is something we need to manually bump?

**Review by tlambert03 — APPROVED** [review]

File: CITATION.cff
Comment (neuromusic) [reply CITATION.cff]: yes. 

looks like it could be dropped? https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md

or is there other tooling the project is using to keep version info sync'd?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:63]: @VolkerH can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:67]: @brisvag can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:71]: @JoOkuma can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:74]: @jaimergp can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:77]: @hectormz

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:81]: @freeman-lab can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:84]: @boonepeter can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:86]: @quantumjot can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:90]: @cgohlke can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:93]: @royerloic can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:96]: @kir0ul can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:98]: @HagaiHargil can you review your name & affiliation?

File: CITATION.cff
Comment (neuromusic) [inline CITATION.cff:101]: @AbigailMcGovern can you review your name & affiliation?

File: CITATION.cff
Comment (brisvag) [inline CITATION.cff]: Your comment was a bit off-line, so I'm making a new one:
```suggestion
- given-names: Lorenzo
  family-names: Gaifas
  affiliation: Gutsche Lab - University of Grenoble
```

File: CITATION.cff
Comment (VolkerH) [reply CITATION.cff:63]: The name "Volker Hilsenstein" is correct.
Current affiliation: 
EMBL Heidelberg, Germany

Thanks Justin!

File: CITATION.cff
Comment (hectormz) [inline CITATION.cff]:
```suggestion
- given-names: Hector
  family-names: Muñoz
  affiliation: University of California, Los Angeles
```

File: CITATION.cff
Comment (hectormz) [reply CITATION.cff:77]: added a new comment for suggested change!

File: CITATION.cff
Comment (HagaiHargil) [inline CITATION.cff:99]:
```suggestion
  family-names: Har-Gil
  affiliation: Tel Aviv University, Israel
```

File: CITATION.cff
Comment (kir0ul) [reply CITATION.cff:96]: It would be:
```yaml
- given-names: Andrea
  family-names: Pierré
  affiliation: Brown University
```
Thanks @neuromusic!

File: CITATION.cff
Comment (boonepeter) [reply CITATION.cff:84]: Looks good thanks!

File: CITATION.cff
Comment (GenevieveBuckley) [reply CITATION.cff:53]: Looks good to me! 
While technically Dask is my affiliation for 11 months in 2021, I was at Monash before and after that, and for longer periods of time - so I think this is fine.

**Comment by Czaki** [comment]
Why there is not attached orcid to persons?

File: CITATION.cff
Comment (JoOkuma) [reply CITATION.cff:71]: It's correct.

File: CITATION.cff
Comment (quantumjot) [reply CITATION.cff:86]: Looks good.

File: CITATION.cff
Comment (jaimergp) [reply CITATION.cff:74]: LGTM! Thanks!

File: CITATION.cff
Comment (brisvag) [reply CITATION.cff:67]: Done in a separate comment.

**Comment by neuromusic** [comment]
> Why there is not attached orcid to persons?

It's not automated, so individuals will need to add them

File: CITATION.cff
Comment (tlambert03) [inline CITATION.cff:14]:
```suggestion
  affiliation: Harvard Medical School
  orcid: "https://orcid.org/0000-0002-2409-0181"
```

😛

File: CITATION.cff
Comment (GenevieveBuckley) [inline CITATION.cff:55]:
```suggestion
  affiliation: Monash University
  orcid:  https://orcid.org/0000-0003-2763-492X 
```

Comment (GenevieveBuckley) [reply CITATION.cff:55]: Turns out I have an ORCID and used it exactly once in my life

File: CITATION.cff
Comment (sofroniewn) [inline CITATION.cff:10]:
```suggestion
  affiliation: Chan Zuckerberg Initiative
  orcid: "https://orcid.org/0000-0002-3426-0914"
```

File: CITATION.cff
Comment (hectormz) [inline CITATION.cff:79]:
```suggestion
- given-names: Hector
  family-names: Muñoz
  affiliation: University of California, Los Angeles
  orcid: https://orcid.org/0000-0001-7851-2549
```

File: CITATION.cff
Comment (brisvag) [inline CITATION.cff:69]:
```suggestion
- given-names: Lorenzo
  family-names: Gaifas
  affiliation: Gutsche Lab - University of Grenoble
  orcid: https://orcid.org/0000-0003-4875-9422
```

**Comment by sofroniewn** [comment]
This is looking great @neuromusic - it would be great to push it over the finish line. Is there stuff that I or @andy-sweet could do to help, either with cleaning up affiliations or the zenodo integration? Is this also something where we could merge the affiliations and then do the zenodo integration in a follow up PR. Thanks!!

**Comment by jni** [comment]
Another Q @neuromusic: do we have a plan for keeping this up to date? Perhaps we need to add this to the reviewer checklist, so that any first-time contributor is prompted to add themselves to the list?

File: CITATION.cff
Comment (Carreau) [inline CITATION.cff]:
```suggestion
  affiliation: '@Quansight-Labs'
  orcid: "http://orcid.org/0000-0002-7636-8632"
```

**Comment by brisvag** [comment]
> Another Q @neuromusic: do we have a plan for keeping this up to date? Perhaps we need to add this to the reviewer checklist, so that any first-time contributor is prompted to add themselves to the list?

We could even have a CI workflow on PRs that fails a new contributor is not in there...

File: CITATION.cff
Comment (neuromusic) [reply CITATION.cff]: dropped the version info & switched to zenodo's "concept" DOI so that this doesn't need to be kept up to date

**Comment by neuromusic** [comment]
OK, a couple of updates...

- I believe affiliations are up to date & accurate

once this PR is merged, Zenodo will refer to this file as the "source of truth" for everything in the file. anticipating this, I've made a couple of changes:

- I changed the DOI to reference Zenodo's "Concept" DOI instead of a DOI associated with a specific release
- I dropped references to "release date"

as for maintaining this going forward, yes, @jni, this will require some additional process to make sure that new contributors are added to this list, either as part of the PR review process or release process. a CI workflow sounds awesome @brisvag , however, this file doesn't maintain either emails or Github handles, so it may be hard to reconcile whether a contributor is in this list or not.

**Comment by neuromusic** [comment]
related to forward maintenance, the author ordering here is what zenodo automatically generated which I think corresponds to volume of contributions in some way (either lines of code or commits? idk.)

curious @jni @sofroniewn what the right author ordering is for the napari community...
- how should the current list be ordered?
- where in the list would a new author be placed?

also, this author list is based on contributions to this repository, which would include documentation, examples, etc.

do contributions to related repositories warrant inclusion as authors here? thinking `napari.github.io`, `npe2`, `magicgui`, etc?

**Comment by sofroniewn** [comment]
Great questions here @neuromusic and my apologies for missing that the ball here was in @jni and my court

> the author ordering here is what zenodo automatically generated which I think corresponds to volume of contributions in some way (either lines of code or commits? idk.)
>
> curious @jni @sofroniewn what the right author ordering is for the napari community...
>
> how should the current list be ordered?
> where in the list would a new author be placed?

i'd be fine sticking with zenodo's default for now to get this merged, but i'm not sure about maintenance 

> also, this author list is based on contributions to this repository, which would include documentation, examples, etc.
>
> do contributions to related repositories warrant inclusion as authors here? thinking napari.github.io, npe2, magicgui, etc?

these are also great questions.

I suggest we merge now and then adopt some process for adding people and reordering based around major releases

File: CITATION.cff
Comment (sofroniewn) [inline CITATION.cff:4]:
```suggestion
message: If you use this software, please cite it using these metadata
```

**Review by sofroniewn — COMMENTED** [review]
Trying to trigger a re-run of the manifest check

File: CITATION.cff
Comment (sofroniewn) [inline CITATION.cff]:
```suggestion
message: If you use this software, please cite it using these metadata.
```

**Comment by sofroniewn** [comment]
@jni is this good to go from your perspective? Can we merge? Test fails are unrelated. I think it would be great to get this in as a start and then press on from there!!

**Comment by jni** [comment]
I agree, let's iterate as we go. Thanks @sofroniewn!

**Comment by sofroniewn** [comment]
Thanks @jni 

@neuromusic I'll merge this tomorrow morning unless I hear otherwise from you or someone else 🚀

**Comment by brisvag** [comment]
For a future PR: would it make sense to add a step to the CI to check if all git authors in history are added in some form? I guess this only works if at least emails are the same though.

**Comment by sofroniewn** [comment]
> For a future PR: would it make sense to add a step to the CI to check if all git authors in history are added in some form? I guess this only works if at least emails are the same though.

yeah i think we need to think about how we keep this list up to date and maybe how we verify it every release. we should probably create a follow up issue to track that discussion. i will merge now though as this is still a great start!!

**Pull request merged**
